src/dbnode/storage/index_insert_queue.go

```diff
@@ -232,21 +258,60 @@ type nsIndexInsertBatch struct {
 	namespace           namespace.Metadata
 	nowFn               clock.NowFn
 	wg                  *sync.WaitGroup
-	shardInserts        []*index.WriteBatch
+	insertsByCPUCore    []*nsIndexInsertsByCPUCore
```

Comment:
Add a comment somewhere here to ensure that neither this or its pointers are modified

Reply:
Will do.